### ISSUE TITLE
perform proper server side validation of email addresses

### ISSUE THREAD

**Issue by lloyd** [mozilla/persona#514]
currently we're not very rigorous in checking the format/length of email addresses, we should add a bit of verification logic given restrictions on email addresses:

```
09:21 < stomlinson> lloyd: according to http://en.wikipedia.org/wiki/Email_address, the local part may be up to 64 characters, the domain part up to 253 characters, but an overall address may be no more than 254 characters.
```

Currently how an overly long email address would fail is unclear.

**Comment by shane-tomlinson** [comment]
@zaach, @lloyd - it looks like there is a function to take care of this now.  Are overly long email addresses handled on the backend?  If so, can this be closed?

**Comment by lloyd** [comment]
yes, addressed by issue #1979
